src/seamless-immutable.js

```diff
@@ -127,7 +127,13 @@
         // Might (validly) be object or array
         newValue = thisHead.setIn(tail, value);
       } else {
-        newValue = arraySetIn.call(immutableEmptyArray, tail, value);
+        var nextHead = tail[0];
+        // If the next path part is a number, then we are setting into an array, else an object.
+        if (!isNaN(parseFloat(nextHead)) && isFinite(nextHead)) {
```

Comment:
Sure, we could use Numberx - something like this?

```js
if (nextHead !== '' && !isNaN(Number(nextHead)) && isFinite(nextHead)) {
```

After reading a bit more about how isFinite works, we could probably actually shorten it to this:

```js
if (nextHead !== '' && isFinite(nextHead)) {
```